lib/ansible/modules/remote_management/redfish/redfish_config.py

```diff
@@ -0,0 +1,228 @@
+      category: Manager
+      command: SetManagerAttributes
+      mgr_attr_name: NTPConfigGroup.1.NTP1
+      mgr_attr_value: "{{ ntpserver1 }}"
+      baseuri: "{{ baseuri }}"
+      user: "{{ user}}"
+      password: "{{ password }}"
+
+  - name: Set Timezone to {{ timezone }} in the OOB Controller
+    redfish_config:
+      category: Manager
+      command: SetManagerAttributes
+      mgr_attr_name: Time.1.Timezone
+      mgr_attr_value: "{{ timezone }}"
+      baseuri: "{{ baseuri }}"
+      user: "{{ user}}"
+      password: "{{ password }}"
+'''
+
+RETURN = '''
+result:
+    description: different results depending on task
+    returned: always
+    type: dict
+    sample: BIOS Attributes set as pending values
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.redfish_utils import RedfishUtils
+
+# More will be added as module features are expanded
+CATEGORY_COMMANDS_ALL = {
+    "Systems": ["SetBiosDefaultSettings", "SetBiosAttributes"],
+    "Manager": ["SetManagerAttributes"],
+}
+
+
+def main():
+    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True),
+            command=dict(required=True, type='list'),
+            baseuri=dict(required=True),
+            user=dict(required=True),
+            password=dict(required=True, no_log=True),
+            mgr_attr_name=dict(),
+            mgr_attr_value=dict(),
+            bios_attr_name=dict(),
+            bios_attr_value=dict(),
+        ),
+        supports_check_mode=False
+    )
+
+    category = module.params['category']
+    command_list = module.params['command']
+
+    # admin credentials used for authentication
+    creds = {'user': module.params['user'],
+             'pswd': module.params['password']}
+
+    # Manager attributes to update
+    mgr_attributes = {'mgr_attr_name': module.params['mgr_attr_name'],
+                      'mgr_attr_value': module.params['mgr_attr_value']}
+    # BIOS attributes to update
+    bios_attributes = {'bios_attr_name': module.params['bios_attr_name'],
+                       'bios_attr_value': module.params['bios_attr_value']}
+
+    # Build root URI
+    root_uri = "https://" + module.params['baseuri']
+    rf_uri = "/redfish/v1"
+    rf_utils = RedfishUtils(creds, root_uri)
+
+    # Check that Category is valid
+    if category not in CATEGORY_COMMANDS_ALL:
+        module.fail_json(msg="Invalid Category: %s" % category)
+
+    # Check that all commands are valid
+    for cmd in command_list:
+        # Fail if even one command given is invalid
+        if cmd not in CATEGORY_COMMANDS_ALL[category]:
+            module.fail_json(msg="Invalid Command: %s" % cmd)
+
+    # Organize by Categories / Commands
+    if category == "Systems":
+        # execute only if we find a System resource
+        result = rf_utils._find_systems_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
```

Comment:
The value in `msg` should be wrapped in `to_native()` to ensure it is the correct string type based on the Python version. `from ansible.module_utils._text import to_native`